### ISSUE TITLE
Custom styles not applied if control is removed & re-added

### ISSUE THREAD

**Issue by derekantrican** [AvaloniaUI/Avalonia#10012]
**Describe the bug**

I have a control with a custom style applied via a class selector. If the control is programmatically removed & re-added, the style is not applied.

![](https://i.imgur.com/WAd6Iv8.gif)

**To Reproduce**
Min repro:

MainWindow.axaml
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" Width="400" Height="450"
        x:Class="AvaloniaMissingStylesDemo.MainWindow"
        Title="AvaloniaMissingStylesDemo">
	<Window.Styles>
		<Style Selector="ComboBox.myClass">
			<Setter Property="ItemTemplate">
				<Setter.Value>
					<DataTemplate>
						<TextBlock Text="{Binding Name}"/>
					</DataTemplate>
				</Setter.Value>
			</Setter>
		</Style>
	</Window.Styles>
	<StackPanel>
		<Button Content="Toggle control" Command="{Binding ToggleControl}" Margin="5"/>
		<Panel x:Name="myPanel">
			<ComboBox Classes="myClass" Width="300" Items="{Binding Items}" Margin="5"/>
		</Panel>
	</StackPanel>
</Window>
```

MainWindow.axaml.cs
```csharp
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Avalonia.Controls;

namespace AvaloniaMissingStylesDemo
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();

            this.DataContext = this;
        }

        public ObservableCollection<MyClass> Items { get; set; } = new ObservableCollection<MyClass>(new List<MyClass>
        {
            new MyClass
            {
                Name = "Foo1",
            },
            new MyClass
            {
                Name = "Foo2",
            },
            new MyClass
            {
                Name = "Foo3",
            },
        });

        Control childControl = null;
        public void ToggleControl()
        {
            Panel panel = this.FindControl<Panel>("myPanel");

            if (panel.Children.Count > 0)
            {
                childControl = panel.Children[0] as Control;
                panel.Children.Clear();
            }
            else
            {
                panel.Children.Add(childControl);
            }
        }

        public class MyClass
        {
            public string Name { get; set; }
        }
    }
}
```

**Expected behavior**
The control still has the classes, when the control is shown again, I expect the style to be re-applied.

**Desktop (please complete the following information):**

- OS: Windows 11
- Version: 0.10.18

**Comment by derekantrican** [comment]
Also tested in 11.0.0-preview4. Issue still repros with above code

**Comment by derekantrican** [comment]
A valid workaround for the time being is to toggle `childControl.IsVisible` as `IsVisible` means the control doesn't take up space (same as `Collapsed` in WPF)

**Comment by grokys** [comment]
Seems to be fixed in `preview5` and current `master`. From a bisect it looks like it was fixed by #9677.
